static/app/components/profiling/FlamegraphZoomView.tsx

```diff
@@ -224,8 +205,26 @@ function FlamegraphZoomView({
           flamegraphRenderer.configView.height
         )
       );
-      setSelectedNode(frame);
+
       setConfigSpaceCursor(null);
+      setSelectedNode(frame);
+
+      newScheduler.draw();
+    });
+
+    newScheduler.on('searchResults', (results: Record<string, FlamegraphFrame>) => {
+      newScheduler.unregisterBeforeFrameCallback(drawRectangles);
+
+      function newDrawRectangles() {
+        if (flamegraphRenderer) {
+          flamegraphRenderer.draw(results);
+        }
+      }
+      newScheduler.registerBeforeFrameCallback(newDrawRectangles);
+
+      newScheduler.onDispose(() =>
+        newScheduler.unregisterBeforeFrameCallback(newDrawRectangles)
+      );
```

Comment:
Yeah, I was debating between that and storing the results. The difference here is that if we pass searchresults to this effect, we end up recreating all the renderers (unless we split each renderer to it's own useEffect init), but here we instead just replace the draw call. I think we need to figure out a good sort of "swap" mechanism here, I'm not even sure about the whole canvasScheduler and how well that will scale tbh

Reply:
If you have some ideas about this, I'd love to hear them. We want a system that allows us to enqueue calls in some order and replace them. Maybe the easiest way here is to just isolate each renderer to it's own effect

Reply:
I took a pass over in https://github.com/getsentry/sentry/pull/31639 where I lifted all the renderers out of `useEffect` using `useMemo`. In that scenario, I think if we used a ref to store the search results to avoid recreating everything, the `.draw` immediately after should accomplish what we want here. It's quite a refactor for an already large PR, so I'm happy to leave this as is and resolve this in a follow up.